### DISCUSSION TITLE
interface ElFormContext 存在的意义

### DISCUSSION THREAD

**Discussion by Otto-J** [element-plus/element-plus#4157] [Q&A] [answered]
对于一个 elForm ，想要手动触发校验，我一般是这样写

```ts
// 对应 el-form ref='formRef'
import type { ElForm } from "element-plus";
const formRef = ref<InstanceType<typeof ElForm>>();

// 当 @click='formSubmit'
const formSubmit = ()=>{
formRef?.validate(valid=>console.log(valid))
}
```

问题一。我一直是这样使用的，感觉挺麻烦，有啥建议吗，这是标准写法吗。要不推个 pr 做个包装？

问题二，在源码 tokens/form.d.ts 里导出了 ElFormContext ，这个对普通用户有作用吗，还是说内部类型不需要导出

**Comment by sxzz** [comment, marked as answer]
关于问题一：似乎就是需要这样写的，没什么问题。如果觉得麻烦可自行封装下类型。

关于问题二：目前 `ElFormContext` 是内部为了 inject el-form 用的，对普通用户应该是没有用处的。忽略即可。

**Reply by Otto-J** [reply]
我觉得这个地方可以推个 pr，暴露一下 ElxxxRef ，比如 ElFormRef，这样就不用让用户来做 ts体操了，不确定是不是个最佳实践，我来个 issues 讨论下？

**Reply by sxzz** [reply]
可以的

**Reply by emojiiii** [reply]
这是个非常棒的主意，或许可以加在 #3113  的计划里面？

**Reply by sxzz** [reply]
赞成

**Comment by Otto-J** [comment]
欢迎大家后续在 #4308 里参与讨论。此贴终结。
